bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -52,7 +53,7 @@ class SchemaUpdateOption(Enum):
 
 
 class Job:
-    def __init__(self, job_id: str,
+    def __init__(self, job_id: Optional[str] = None,
```

Comment:
non-actionable comment: something about making this `job_id` optional feels off to me. Like, `Job.insert_via_query` totally makes sense to me, since that's how Google has it defined, but the fact that we might have `Job()` instances which aren't actually pointing to real jobs... I dunno, I wonder if it'd be hard to reason about?

Might point to aiming towards future refactoring efforts -- eg. it feels to me like perhaps this is a instance method vs class method thing? ie. doing `Job.insert_via_query(...)` makes sense, rather than having to first initialize the `Job` class. That messes with the `self.project()` calls, so it definitely doesn't fit the current system, but I wonder how best to make it work. Maybe the official Google libs have a good answer here?

Reply:
Well, this is in fact a sneak peak onto the abyss of complexity that google library is: https://github.com/googleapis/python-bigquery/blob/master/google/cloud/bigquery/client.py#L2778-L2809

In order to run query `google-cloud-bigquery` _generates_ job_id locally, creates a `JobReference` object that references nothing real at this point, then `QueryJob` object, links them together and only after this a call to the server is made.